src/include/txn.i

```diff
@@ -442,6 +442,45 @@ __wt_txn_op_apply_prepare_state(
 	WT_REF_SET_STATE(ref, previous_state);
 }
 
+/*
+ * __wt_txn_op_delete_commit_apply_timestamps --
+ *	Apply the correct start and durable timestamps to any
+ *	updates in the page del update list.
+ */
+static inline void
+__wt_txn_op_delete_commit_apply_timestamps(
+    WT_SESSION_IMPL *session, WT_REF *ref)
+{
+	WT_TXN *txn;
+	WT_UPDATE **updp;
+	wt_timestamp_t ts;
+	uint32_t previous_state;
+
+	txn = &session->txn;
+	ts = txn->commit_timestamp;
+
+	/*
+	 * Lock the ref to ensure we don't race with eviction freeing the page
+	 * deleted update list or with a page instantiate.
+	 */
+	for (;; __wt_yield()) {
+		previous_state = ref->state;
+		WT_ASSERT(session, previous_state != WT_REF_READING);
+		if (previous_state != WT_REF_LOCKED && WT_REF_CAS_STATE(
+		    session, ref, previous_state, WT_REF_LOCKED))
+			break;
+	}
+
+	for (updp = ref->page_del->update_list;
+	    updp != NULL && *updp != NULL; ++updp) {
+		(*updp)->start_ts = ts;
```

Comment:
it would be more readable to replace ts with txn->commit_timestamp OR, save both txn->durable_timestamp and txn->commit_timestamp into local variables and assign to each update.